### ISSUE TITLE
readline() returns null instead of false on EOF

### ISSUE THREAD

**Issue by tstarling** [facebook/hhvm#4776]
readline() should return false on EOF, and the C++ code implies that this was intended by the author:

```
static Variant HHVM_FUNCTION(readline, const String& prompt) {
  auto result = readline(prompt.data());
  if (result == nullptr) {
    return false;
```

But in the systemlib specifies the return value as "string" not "mixed", which I guess causes non-string values to be converted to null. Observed on HHVM 3.5.0.

**Comment by mssola** [comment]
This has been fixed with 876679350fde83b73433c01e8e6329d2a41321bd no ?

**Comment by paulbiss** [comment]
Looks like it.
